Translate into English of user-guide (#50)
opst/knitfab: docs/02.user-guide/user-guide.en.md

resource:
    cpu: 1
    memory: 1Gi
```

The meanings of each key are as follows:

- `tag`: Tags that are commonly set for Data and Plan inputs and outputs (multiple tags can be set; default is empty)
    - In the example above, all registered Data and created Plans will automatically have the Tag `project:some-project` set.
- `resource`: Required computing resources for creating Plan definition templates.
    - In this example, the project defaults to using 1 CPU core and 1 GB of memory for processing this Plan.
    - More details about this item will be covered in the Plan Definition section.

This file is intended to be shared among the team.
If you are version controlling your machine learning project with git or similar tools, it is recommended to include this file in version control.

Managing Data
-----

Data refers to directories with tags that contain the results of machine learning tasks. Users can also create Data by registering it with Knitfab.

The created Data in Knitfab is compared with the inputs of each Plan and if there is an executable Plan, it is executed as a Run.

This chapter discusses the handling of this Data.

### Registering Data

> If you want to use a collection of files, such as a dataset, as input for a machine learning task, you can register it as Data in Knitfab.

Reviewer: ```suggestion
When you want to use a collection of files, such as a dataset, as input for a machine learning task, you can register it as Data in Knitfab.
```

If は「（起きるかどうかわからないが、）もしおきたら」というニュアンスで、when は「（きっとそうなるだろうが、）もしそうなれば」というニュアンスだそうです。
ユーザはデータセットを登録したいに決まっているので、when でいいでしょう、

```json
{
    "knitId": "a0f5911d-5ac5-4035-8c91-9946b02adc70",
    "tags": [
        "knit#id:a0f5911d-5ac5-4035-8c91-9946b02adc70",
        "knit#timestamp:2024-03-06T09:25:16.586+00:00",
        "project:some-project",
        "your:tag"
    ],
    "upstream": {
        "path": "/upload",
        "tags": [],
        "run": {
            "runId": "a5c1106c-122f-416c-b4f4-e6dd03af212c",
            "status": "done",
            "updatedAt": "2024-03-06T09:25:16.586+00:00",
            "plan": {
                "planId": "fb9b7087-9fe7-43c3-9302-a35488c85ffb",
                "name": "knit#uploaded"
            }
        }
    },
    "downstreams": [],
    "nomination": []
}
```

- `knitId`: This is the ID that identifies this Data.
- `tags`: It shows the Tags that are set for this Data.
    - It includes automatically assigned system-generated Tags (with keys starting with `knit#`) and user-defined Tags.
    - Additionally, if there are any Tags set in `knitenv`, they will also be included.
- `upstream`: It shows the output where this Data was written and information about the corresponding Run.

>     - In this case, it displays a special Run of a Plan that accepts the upload of Data.

Reviewer: ```suggestion
    - In this case, it displays a Run of a special  Plan that accepts the upload of Data.
```

                        "message": "Completed"
                    },
                    "plan": {
                        "planId": "6dbfed46-f9fb-4e4e-aadd-2a5c42d26913",
                        "image": "localhost:30503/Knitfab-first-validation:v1.0"
                    }
                }
            }
        ],
        "nomination": [
            {
                "path": "/in/dataset",
                "tags": [
                    "mode:test",
                    "project:first-Knitfab",
                    "type:dataset"
                ],
                "plan": {
                    "planId": "6dbfed46-f9fb-4e4e-aadd-2a5c42d26913",
                    "image": "localhost:30503/Knitfab-first-validation:v1.0"
                }
            }
        ]
    }
]
```

The structure of each element is the same as described in the "Registering Data" section. In this example, since the Data has been used as input in other runs, the `downstreams` and `nomination` also contain the details.

##### Command-line options

- `-t key:value`, `--tag key:value`: Search for Data with the specified Tag.
    - If multiple Tags are specified, only the Data with all the Tags will be searched.

> If `-t` is not specified at all, all Data will be searched.

Reviewer: ```suggestion
If `-t` is not specified at all, all Data will be shown.
```

#### Downloading the Data Itself

You can not only view metadata but also download the actual Data itself.

You may often want to download model parameters or evaluation results.

```
knit data pull -x KNIT_ID path/to/dest
```

This command writes the contents of the Data identified by the ID `KNIT_ID` to the directory `path/to/dest/${KNIT_ID}`.

##### Command-line options

- `-x`: Download and extract the Data.
    - If not specified, the Data"will be downloaded as a `tar.gz` compressed file to `path/to/dest/${KNIT_ID}.tar.gz`.

### Changing Data Tags

You can change the Tags of already registered Data.

For example,

- If there is a Tag `deployed:true` that indicates "deployed", you may want to move it to other Data.
- If you no longer want a certain Data to be used as input for a Run, you may want to remove the Tag.
- If you want to use a certain Data in a specific Plan, you may want to add the Tag.

This is useful in such cases.

To add Tag, you can use the following command:

```
knit data tag --add new:tag KNIT_ID
```

> , and This will add the "new:tag" Tag to the Data identified by the ID `KNIT_ID`.

Reviewer: ```suggestion
and This will add the "new:tag" Tag to the Data identified by the ID `KNIT_ID`.
```

logs:
  - "project:some-project"
  - "type:log"
```

- `image`: Specifies "which image" for the processing.
    - The container image name is specified with the tag.
        - The image name follows the structure of `[host:port/][repository/]name:tag`.
    - If you want to use an image registered in the Knitfab image registry,
        - Set the host to `localhost`.
        - The port is typically `30503` for a standard installation. Please check with the administrator who performed the installation for details.
- `inputs`: Specifies "what inputs" to assign.
    - Each input is a combination of `path` and `tags`. Multiple inputs can be specified.
        - In the example, there are two inputs: `/in/dataset` and `/in/params`.
    - For each input,
        - `path` indicates the directory in the container where the input is provided when executing this task.
        - `tags` indicate what Data can be assigned to this input.
            - You can specify "system Tags" such as `knit#id` and `knit#timestamp`. This can be useful if you want to do something specific after identifying the Data.
    - At least one input is required. And each input must have at least one "Tag" set.

> - `outputs`: Specifies "what outputs" can be obtained.

Reviewer: ```suggestion
- `outputs`: Specifies "what outputs" will be obtained.
```

"can be" だと、「得られうるのか」みたいな意味になるように思われます。つまり、得られないことがある、という含意が示されてしまっているように思われるのです。

    "image": "localhost:30503/Knitfab-first-validation:v1.0",
    "inputs": [
        {
            "path": "/in/dataset",
            "tags": [
                "mode:test",
                "project:first-Knitfab",
                "type:dataset"
            ]
        },
        {
            "path": "/in/model",
            "tags": [
                "project:first-Knitfab",
                "type:model"
            ]
        }
    ],
    "outputs": [],
    "log": {
        "Tags": [
            "project:first-Knitfab",
            "type:log",
            "type:validation"
        ]
    },
    "active": true,
    "resources": {
        "cpu": "1",
        "memory": "1Gi"
    }
}
```

Apart from the difference between JSON and YAML, the structure of a Plan definition is the same.

If you don't know the ID of the Plan you are interested in, you can also search for it.

If you know the image name, you can use the following command:

```
knit plan find --image HOST:PORT/IMAGE:TAG
```

If you know the Tags used in the inputs or outputs, you can use the following:

```
knit plan find --in-tag KEY:VALUE
```

```
knit plan find --out-tag KEY:VALUE
```

> If you want to limit the search to active Plans", use the following command:

Reviewer: ```suggestion
If you want to limit the search to active Plans, use the following command:
```

```
knit plan find --active yes
```

> These search conditions can be specified simultaneously, and only the Plans that meet all the conditions will be matched in the search.

Reviewer: ```suggestion
These search conditions can be specified simultaneously, and only the Plans that meet all the conditions will be shown.
```

- `--image IMAGE[:TAG]`: Searches based on the image of the Plan.
    - `TAG` can be omitted. In that case, the search ignores the tags of the image.
    - The `IMAGE` part also matches the hostname or port of the registered image in the Plan definition. It does not perform partial matching.
- `-i KEY:VALUE`, `--in-tag KEY:VALUE`: Limits the search to those with inputs that have the specified Tags.
    - If this flag is specified multiple times, it only searches for those with inputs that have all the specified Tags.
- `-o KEY:VALUE`, `--out-tag KEY:VALUE`: Limits the search to those with outputs that have the specified Tags.
    - If this flag is specified multiple times, it only searches for those with outputs that have all the specified Tags".

### Activating and deactivating a Plan

If you no longer need a Plan, you can deactivate it:

```
knit plan active no PLAN_ID
```

By executing this command, the Plan identified by the specified `PLAN_ID` will be deactivated.
Any Run based on this Plan that has not started yet will be prevented from executing.

You can also reactivate a Plan that was deactivated for some reason.

```
knit plan active yes PLAN_ID
```

> This command will re-execute any Run based on the Plan with the specified `PLAN_ID` that was previously prevented from executing.

Reviewer: ```suggestion
This command will re-execute any prevended Run based on the Plan with the specified `PLAN_ID`.
```

Author: preventの綴りを修正した、こちらでコミットしますね。
```suggestion
This command will re-execute any prevented Run based on the Plan with the specified `PLAN_ID`.
```

### Updating the compute resources required by a Plan

> Let's say you have executed a computational task and found that there is an excess or shortage of compute resources specified in the "Plan" definition. This can be a problem, especially if there is a shortage of resources, as the computation cannot be executed.

Reviewer: ```suggestion
Let's say you have executed a computational task and found that there is an excess or shortage of compute resources specified in the Plan definition. This can be a problem, especially if there is a shortage of resources, as the computation cannot be executed.
```

The following output is expected:

```json
[
    {
        "runId": "3cb1b091-01ad-41b1-acac-3f042f9df97c",
        "status": "starting",
        "updatedAt": "2024-03-07T06:28:31.59+00:00",
        "plan": {
            "planId": "6dbfed46-f9fb-4e4e-aadd-2a5c42d26913",
            "image": "localhost:30503/Knitfab-first-validation:v1.0"
        },
        "inputs": [
            {
                "path": "/in/dataset",
                "tags": [
                    "mode:test",
                    "project:first-Knitfab",
                    "type:dataset"
                ],
                "knitId": "87038a1d-467c-45d4-9cc7-876473f3fc04"
            },
            {
                "path": "/in/model",
                "tags": [
                    "project:first-Knitfab",
                    "type:model"
                ],
                "knitId": "46173435-d55a-46ae-8300-11560ca371d9"
            }
        ],
        "outputs": [],
        "log": {
            "Tags": [
                "project:first-Knitfab",
                "type:log",
                "type:validation"
            ],
            "knitId": "9bb07d29-ab79-4ec0-a233-ec000e418261"
        }
    }
]
```

> > If multiple Runs are found, the elements will be in an array.

Reviewer: ```suggestion
> If multiple Runs are found, more elements will be in an array.
```
